toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -43,6 +43,37 @@ var (
 	ToolVersion = ""
 )
 
+type ImageCustomizer struct {
```

Comment:
re. #1. The refactoring of the code into three stages (convert to writeable, customize, convert to final image) is related to this change as it makes the code more readable now we have the iso scenarios intermingled with customizations. The core of mic customization code has not actually changed at all (diff looks bad because of the parameters) - it has its own function right now. The parts that changed a bit is the 'convert to writeable' and the 'convert to final image'.

re. #2. I debated between the two approaches, Chris. But having flat functions here makes it look very unmanageable - especially with the increase in the number of parameters. I leaned towards centralizing the core and derived parameters in one place so that it's more consistent as well as easier to pass to consumers.
I agree that there's a chance of having side effects, maybe I should just make this structure a read-only parameter and pass it into the various functions. All its state is setup when it is created and should not change after that (also, intentionally - since I agree re. no side effects).

Reply:
If you want to create a parameters struct, I'm ok with that.

Reply:
sadly, no read-only marking in golang. Passed it as a parameters struct.
pending.